pkg/controller/database/rdsinstance.go

```diff
@@ -204,18 +205,16 @@ func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 	modify := rds.GenerateModifyDBInstanceInput(meta.GetExternalName(cr), patch)
 	var conn managed.ConnectionDetails
 	if cr.Spec.ForProvider.MasterPasswordSecretRef != nil {
```

Comment:
I think we can skip the `if cr.Spec.ForProvider.MasterPasswordSecretRef != nil` here since we do it in `GetPassword` anyway. The `if changed` condition covers the case where there is no change.